src/components/Filter/GroupedEnumFilter.tsx

```diff
@@ -46,17 +46,21 @@ export const GroupedEnumFilter = ({
 
   const deleteGroup = (groupId: string): void =>
     onSelectedEnumIdsChange(
-      selectedEnumIds.filter((enumId) => id2enum?.[enumId]?.groupId !== groupId),
+      selectedEnumIds
+        .filter((id) => id2enum[id])
+        .filter((enumId) => id2enum?.[enumId]?.groupId !== groupId),
```

Comment:
I know the extra optional chaining doesn't cause problems, but it does get distracting with alternating use.